api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -676,6 +683,45 @@ func (m *AzureManagedControlPlane) validateOIDCIssuerProfileUpdate(old *AzureMan
 	return allErrs
 }
 
+// validateMarketplaceExtensionsUpdate validates update to Marketplace extensions.
+func (m *AzureManagedControlPlane) validateMarketplaceExtensionsUpdate(old *AzureManagedControlPlane) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if m.Spec.MarketplaceExtensions != nil && old.Spec.MarketplaceExtensions != nil {
+		for i, extension := range m.Spec.MarketplaceExtensions {
+			if extension.Name != old.Spec.MarketplaceExtensions[i].Name {
```

Comment:
This validation looks like it expects the list of `Spec.MarketplaceExtensions` to be in the same order since otherwise the name check will fail if you swap the order of two entries. Do you think it'd make sense to treat different permutations as identical?

Reply:
Yup I realized the error in the logic there and redid the logic to use a map: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4360/files/07d7bac254be26620489fcdea39fe9d8325aad35..a4281810698097954f1e1332c922e0a6ef474225#diff-fba23483604b01adf9ec7f060a2e74fc7762cf8b9bb166451e3275389be96270R692